[DOCS-13462] Add generic amazon s3 doc (#35354)
DataDog/documentation: content/en/observability_pipelines/destinations/amazon_s3.md

---
title: Amazon S3 Destination
description: Learn how to configure the Amazon S3 destination.
disable_toc: false
products:
- name: Logs
  icon: logs
  url: /observability_pipelines/configuration/?tab=logs#pipeline-types
---

{{< product-availability >}}

## Overview

Use the Amazon S3 destination to send logs in JSON or Parquet format to Amazon S3. For Parquet, the Observability Pipelines Worker automatically generates a Parquet schema based on the batch file the events are being written to, which means the schema can vary between batches.

You can also [route logs to Snowflake using the Amazon S3 destination](#route-logs-to-snowflake-using-the-datadog-archives-destination).

> **Note**: If you want to send logs to a S3 bucket and be able to [rehydrate][1] them for analysis and investigation in Datadog, use the [Datadog Archives][2] destination.

Reviewer: ```suggestion
**Note**: If you want to send logs to an S3 bucket, and later be able to [rehydrate][1] them for analysis and investigation in Datadog, use the [Datadog Archives][2] destination.
```

1. Enter your S3 bucket name. If you configured Log Archives, it's the name of the bucket you created earlier.
1. Enter the AWS region the S3 bucket is in.
1. (Optional) Enter the key prefix.
    - Prefixes are useful for partitioning objects. For example, you can use a prefix as an object key to store objects under a particular directory. If using a prefix for this purpose, it must end in `/` to act as a directory path; a trailing `/` is not automatically added.
      - See [template syntax][4] if you want to route logs to different object keys based on specific fields in your logs.
    - **Notes**:
        - Datadog recommends that you start your prefixes with the directory name and without a lead slash (`/`). For example, `app-logs/` or `service-logs/`.
        - Do **not** use the same S3 prefix as a [Datadog Archives][2] destination. The Amazon S3 destination writes files in a different format and having both file types in the same prefix can result in rehydration issues.
1. Select the storage class for your S3 bucket in the **Storage Class** dropdown menu.
1. Select the encoding you want to use in the **Encoding** dropdown menu (**JSON** or **Parquet**).

>     - **Note**: For **Parquet**, the Observability Pipelines Worker automatically generates a Parquet schema based on the batch file the events are being written to, which means the schema can vary between batches.

Reviewer: ```suggestion
    - **Note**: For **Parquet**, the schema is generated per batch and can vary. See [Automatically generated Parquet schema](#automatically-generated-parquet-schema).
```

1. Enter the ARN of the IAM role you want to assume.
    - **Note:** The [user or role you created earlier](#set-up-an-iam-policy-that-allows-workers-to-write-to-the-s3-bucket) must have permission to assume this role so that the Worker can authenticate with AWS.
1. (Optional) Enter the assumed role session name and external ID.

#### Buffering

{{% observability_pipelines/destination_buffer %}}

### Set secrets

{{% observability_pipelines/set_secrets_intro %}}

{{< tabs >}}
{{% tab "Secrets Management" %}}

There are no secret identifiers to configure.

{{% /tab %}}

{{% tab "Environment Variables" %}}

{{% observability_pipelines/destination_env_vars/datadog_archives_amazon_s3 %}}

{{% /tab %}}
{{< /tabs >}}

## Route logs to Snowflake using the Amazon S3 destination

You can route logs from Observability Pipelines to Snowflake using the Amazon S3 destination by configuring Snowpipe in Snowflake to automatically ingest those logs. To set this up:

> 1. [Set up a pipeline][5] to use Amazon S3 as the log destination. When logs are collected by Observability Pipelines, they are written to an S3 bucket using the same configuration detailed in [Set up the destination for your pipeline](#set-up-the-destination-for-your-pipeline), which includes AWS authentication, region settings, and permissions.

Reviewer: ```suggestion
You can route logs from Observability Pipelines to Snowflake using the Amazon S3 destination by configuring Snowpipe in Snowflake to automatically ingest those logs. When logs are collected by Observability Pipelines, they are written to an S3 bucket. To set this up:
1. [Set up a pipeline][5] to use Amazon S3 as the log destination. Use the configuration detailed in [Set up the destination for your pipeline](#set-up-the-destination-for-your-pipeline).
```

Author: This is a great point. I moved the explanatory text in step 2 into the intro paragraph as well.
